core/model/modx/processors/resource/undelete.class.php

```diff
@@ -93,6 +95,50 @@ public function removeLock() {
     }
 
     /**
+     * Get the IDs of all the children of the Resource
+     * @return array
+     */
+    public function getChildrenIds() {
+        $this->children = array();
+        $this->getChildren($this->resource);
+
+        /* prepare children ids for invokeEvents */
+        $childrenIds = array ();
+        /** @var modResource $child */
+        foreach ($this->children as $child) {
+            $childrenIds[] = $child->get('id');
+        }
+        return $childrenIds;
+    }
+
+    /**
+     * Helper method for getChildrenIds for getting Children recursively
+     *
+     * @see getChildrenIds
+     * @param modResource $parent
+     * @return void
+     */
+    protected function getChildren(modResource $parent) {
+        $childResources = $parent->getMany('Children');
```

Comment:
I don't think this is correct, as you'll get all children by this, not only those that were un-deleted.